### PR TITLE
Cart wishlist management

### PR DESCRIPTION
**Pull request by varun-Tad**
Files to review:
1. Cards -> ProdListingCard.js
2. ProdListingPage->ProdListingPage.js
3. CardPage->CartDetails.js
4. reducerFnOne.js
5. WishListDetails.js

Please review add to Cart/Add to wishlist functionality from product page,Cart Page and Wishlist Page.
Please ignore the final purchase card in Cart Page as it has not been fully implemented.Also there are a few design errors overall. will work on them and raise a PR next time 

Preview: https://jolly-cassata-eefd86.netlify.app/

### CODE REVIEW

File: dressup/src/components/Cards/ProdListingCard.js [hunk 1]
```diff
@@ -3,19 +3,20 @@ import "./ProdListingCard.css";
...
 import { FaHeart } from "react-icons/fa";
 import { FaShoppingCart } from "react-icons/fa";
-
+import { useWishCart } from "../../pages/CartPage/WishCart-context";
+const message = (msg) => {
+  console.log(msg);
+};
 function ProdListingCard(props) {
+  const { stateOne, dispatchOne } = useWishCart();
   return (
```
Comment (Sahil25061999) [inline dressup/src/components/Cards/ProdListingCard.js:12]: it will be good if you destructure the props.

**Review by Sahil25061999 — APPROVED** [review]
overall looks good, just a ui suggestion it will be nice if there is some kind of margin between the add to cart and add to wishlist button in the product listing page, and also reduce the padding a bit. And when clicked on the buttons, some kind of feedback can be provided so that the user knows the item is added to the respective list. You can also fix the footer at the bottom of the page.

File: dressup/src/components/Cards/ProdListingCard.js [hunk 2]
```diff
@@ -26,12 +27,27 @@ function ProdListingCard(props) {
           </p>
         </div>
         <div className="button-text-contain">
-          <button type="button" class="btn btn-success btns btn-one">
-            Buy Now
-          </button>
-          <button type="button" className="btn btn-success btns btn-two">
+          <button
+            class="btn btn-success btns btn-one"
```
Comment (Ankur9669) [inline dressup/src/components/Cards/ProdListingCard.js:31]: Use className instead of class

File: dressup/src/components/Cards/ProdListingCard.js [hunk 3]
```diff
@@ -26,12 +27,27 @@ function ProdListingCard(props) {
...
+            onClick={() => {
```
Comment (Ankur9669) [inline dressup/src/components/Cards/ProdListingCard.js:32]: Suggestion: Move this function to a seprate function you can call that as handleButtonClick or anything

File: dressup/src/components/Cards/ProdListingCard.js [hunk 4]
```diff
@@ -26,12 +27,27 @@ function ProdListingCard(props) {
...
+              stateOne.Cart.some((e) => e.id === props.prodData.id)
+                ? message("already exists")
+                : dispatchOne({ type: "addProToCart", value: props.prodData });
+            }}
+          >
             <FaShoppingCart></FaShoppingCart> Add to Cart
           </button>
+
+          <button
+            class="btn btn-success btns btn-two"
```
Comment (Ankur9669) [inline dressup/src/components/Cards/ProdListingCard.js:42]: Use className

File: dressup/src/pages/CartPage/BillDetails/BillDetails.css [hunk 5]
```diff
@@ -40,3 +40,15 @@
   padding: 1rem 0 1rem 1rem;
   margin-bottom: 1rem;
 }
+
+.check-container button {
+  background-color: var(--color-primary-darker);
+  color: var(--color-white);
+  border: none;
+  font-size: 16px;
```
Comment (Ankur9669) [inline dressup/src/pages/CartPage/BillDetails/BillDetails.css:48]: If possible use rems here

File: dressup/src/pages/CartPage/CartDetails/CartDetails.js [hunk 6]
```diff
@@ -1,34 +1,59 @@
...
+                dispatchOne({ type: "removeFromCart", value: props.CartData })
+              }
+            >
+              <FaShoppingCart></FaShoppingCart> Remove from Cart
             </button>
-            <button type="button" className="btn btn-success btns cart-btn">
-              Add to WishList
+            <button
+              className="cart-btn btn btn-success btns buyNow-btn"
+              onClick={() => {
```
Comment (Ankur9669) [inline dressup/src/pages/CartPage/CartDetails/CartDetails.js:50]: Same you can move this function to seprate function

File: dressup/src/pages/CartPage/CartListingPage/CartListingPage.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import CartDetails from "../CartDetails/CartDetails";
+import { useWishCart } from "../WishCart-context";
+
+const CartListingPage = () => {
+  const { stateOne } = useWishCart();
+
+  return (
+    <div>
+      {stateOne.Cart.map((item) => {
+        return <CartDetails CartData={item}></CartDetails>;
```
Comment (Ankur9669) [inline dressup/src/pages/CartPage/CartListingPage/CartListingPage.js:10]: fix: use key here

File: dressup/src/pages/CartPage/reducerFnOne.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
+const addFunc = (state, item) => {
+  const objIndex = state.findIndex((obj) => obj.id === item.id);
+
+  state[objIndex].quantity = item.quantity + 1;
+
+  state[objIndex].newPrice = item.cost * state[objIndex].quantity;
+
```
Comment (Ankur9669) [inline dressup/src/pages/CartPage/reducerFnOne.js:7]: Fix: Here you are mutating the state directly

File: dressup/src/pages/WishlistPage/WishListingPage/WishListingPage.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import WishlistDetails from "../WishlistDetails/WishlistDetails";
+import "./WishListingPage.css";
+
+const WishListingPage = () => {
+  const { stateOne } = useWishCart();
+
+  return (
+    <div className="wish-container">
+      {stateOne.Wishlist.map((item) => {
+        return <WishlistDetails WishData={item}></WishlistDetails>;
```
Comment (Ankur9669) [inline dressup/src/pages/WishlistPage/WishListingPage/WishListingPage.js:11]: Use key here

**Review by Ankur9669 — APPROVED** [review]
The code looks fine to me although i have some UI suggestions
image

On hovering the card it is going on the footer

When we click on move to cart button the wishlist gets empty, but we should have out product in the wishlist and we should add the product in cart.

image

The Heading is going on right side

image


There is a lot of empty space beetween navbar and the products

File: dressup/src/components/Cards/ProdListingCard.js (hunk 1 above)
Comment (varun-Tad) [reply dressup/src/components/Cards/ProdListingCard.js:12]: Okay will work on this

**Pull request merged**